cdc/kv/shared_client.go

```diff
@@ -540,119 +548,129 @@ func (s *SharedClient) divideAndScheduleRegions(
 			}
-			verID := tikv.NewRegionVerID(region.Id, region.RegionEpoch.ConfVer, region.RegionEpoch.Version)
-			sri := newSingleRegionInfo(verID, partialSpan, nil)
-			sri.requestedTable = requestedTable
-			s.scheduleRegionRequest(ctx, sri)
 
-			nextSpan.StartKey = region.EndKey
+			verID := tikv.NewRegionVerID(regionMeta.Id, regionMeta.RegionEpoch.ConfVer, regionMeta.RegionEpoch.Version)
+			regionInfo := newRegionInfo(verID, intersectantSpan, nil, subscribedTable)
+
+			// Schedule a region request to subscribe the region.
+			s.scheduleRegionRequest(ctx, regionInfo)
+
+			nextSpan.StartKey = regionMeta.EndKey
+			// If the nextSpan.StartKey is larger than the subscribedTable.span.EndKey,
+			// it means all span of the subscribedTable have been requested. So we return.
 			if spanz.EndCompare(nextSpan.StartKey, span.EndKey) >= 0 {
 				return nil
 			}
 		}
 	}
 }
 
-func (s *SharedClient) scheduleRegionRequest(ctx context.Context, sri singleRegionInfo) {
-	handleResult := func(res regionlock.LockRangeResult) {
+// scheduleRegionRequest locks the region's range and send the region to regionCh,
+// which will be handled by handleRegionRequest.
+func (s *SharedClient) scheduleRegionRequest(ctx context.Context, region regionInfo) {
+	handleLockResult := func(res regionlock.LockRangeResult) {
 		switch res.Status {
 		case regionlock.LockRangeStatusSuccess:
-			sri.lockedRange = res.LockedRange
-			lockTime := time.Since(sri.lockedRange.Created).Milliseconds()
+			region.lockedRange = res.LockedRange
+			lockTime := time.Since(region.lockedRange.Created).Milliseconds()
 			s.metrics.regionLockDuration.Observe(float64(lockTime))
 			select {
-			case s.regionCh.In() <- sri:
+			case s.regionCh.In() <- region:
 			case <-ctx.Done():
 			}
 		case regionlock.LockRangeStatusStale:
 			for _, r := range res.RetryRanges {
-				s.scheduleRangeRequest(ctx, r, sri.requestedTable)
+				s.scheduleRangeRequest(ctx, r, region.subscribedTable)
 			}
 		default:
 			return
 		}
 	}
 
-	rangeLock := sri.requestedTable.rangeLock
-	res := rangeLock.LockRange(ctx, sri.span.StartKey, sri.span.EndKey, sri.verID.GetID(), sri.verID.GetVer())
+	rangeLock := region.subscribedTable.rangeLock
+	res := rangeLock.LockRange(ctx, region.span.StartKey, region.span.EndKey, region.verID.GetID(), region.verID.GetVer())
 	if res.Status == regionlock.LockRangeStatusWait {
 		res = res.WaitFn()
 	}
-	handleResult(res)
+	handleLockResult(res)
 }
 
 func (s *SharedClient) scheduleRangeRequest(
 	ctx context.Context, span tablepb.Span,
-	requestedTable *requestedTable,
+	subscribedTable *subscribedTable,
 ) {
 	select {
-	case s.requestRangeCh.In() <- rangeTask{span: span, requestedTable: requestedTable}:
+	case s.rangeTaskCh.In() <- rangeTask{span: span, subscribedTable: subscribedTable}:
 	case <-ctx.Done():
 	}
 }
 
-func (s *SharedClient) handleErrors(ctx context.Context) error {
+func (s *SharedClient) handleError(ctx context.Context) error {
```

Comment:
Maybe `handleErrors` is better. It's a goroutine, not a single call for one region.